File: tests/TUnit.TestProject/ClassConstructorTest.cs
namespace TUnit.TestProject;

[ClassConstructor<DependencyInjectionClassConstructor>]
public class ClassConstructorTest(DummyReferenceTypeClass dummyReferenceTypeClass)
{
    public DummyReferenceTypeClass DummyReferenceTypeClass { get; } = dummyReferenceTypeClass;

    [Test]
    public void Test()
    {
    }
}

File: tests/TUnit.TestProject.FSharp/ClassConstructorTest.fs
namespace TUnit.TestProject

open TUnit.Core

[<ClassConstructor(typeof<DependencyInjectionClassConstructor>)>]
type ClassConstructorTest(dummyReferenceTypeClass: DummyReferenceTypeClass) =

    member _.DummyReferenceTypeClass = dummyReferenceTypeClass

    [<Test>]
    member _.Test() = ()

